### PR TITLE
[VL] Fix ORC related failed UT

### PR DESCRIPTION
**Pull request by chenxu14**

### CODE REVIEW

File: gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala [hunk 1]
```diff
@@ -403,13 +403,11 @@ class VeloxTestSettings extends BackendTestSettings {
   enableSuite[GlutenOrcPartitionDiscoverySuite]
     .exclude("read partitioned table - normal case")
     .exclude("read partitioned table - with nulls")
-    .disableByReason("Blocked by ORC Velox upstream not ready")
```
Comment (rui-mo) [inline gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:406]: Some ORC tests are disabled with the message `ReaderFactory is not registered for format orc` like [VeloxTestSettings](https://github.com/oap-project/gluten/blob/main/gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala#L361-L368), could you also enable them in this PR?

**Review by rui-mo — COMMENTED** [review]
We just merged upstream_velox branch into main. Could you change this PR to target on the main branch?

File: gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala (hunk 1 above)
Comment (chenxu14) [reply gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:406]: OK, will try it latter

**Comment by rui-mo** [comment]
Please change this PR to target on the main branch, thanks.

File: gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala [hunk 2]
```diff
@@ -358,8 +356,6 @@ class VeloxTestSettings extends BackendTestSettings {
     .exclude("Return correct results when data columns overlap with partition " +
       "columns (nested data)")
     .exclude("SPARK-31116: Select nested schema with case insensitive mode")
-    // ReaderFactory is not registered for format orc.
-    .exclude("SPARK-15474 Write and read back non-empty schema with empty dataframe - orc")
```
Comment (rui-mo) [inline gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:362]: Below tests are all about ORC issue, thanks.
https://github.com/oap-project/gluten/blob/main/gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala#L361-L368

**Comment by rui-mo** [comment]
The failed tests `nested column: Count(nested sub-field) not push down` seem not relevant to ORC. We can disable them in this PR, and I will take a look further.

**Comment by chenxu14** [comment]
> The failed tests `nested column: Count(nested sub-field) not push down` seem not relevant to ORC. We can disable them in this PR, and I will take a look further.

disable GlutenOrcV2AggregatePushDownSuite, Velox don’t support ORC v2 yet.

File: cpp/velox/compute/VeloxBackend.cc [hunk 3]
```diff
@@ -239,6 +239,7 @@ void VeloxBackend::init(const std::unordered_map<std::string, std::string>& conf
   registerConnector(hiveConnector);
   velox::parquet::registerParquetReaderFactory();
   velox::dwrf::registerDwrfReaderFactory();
+  velox::dwrf::registerOrcReaderFactory();
```
Comment (rui-mo) [inline cpp/velox/compute/VeloxBackend.cc:242]: PR #3445 is going to remove the registration for Parquet and DWRF reader. I'm wondering if the registration for ORC reader can also be done in Velox, then this change can also be removed.

Comment (chenxu14) [reply cpp/velox/compute/VeloxBackend.cc:242]: OK, Adjusted this in PR [417](https://github.com/oap-project/velox/pull/417)
After that merge, I can submit a new PR base on main branch

**Comment by JkSelf** [comment]
@chenxu14 We have enabled spark 3.4 in Gluten. Can you help to also enable the orc unit test in Spark 3.4? Thanks.

**Comment by chenxu14** [comment]
move work to #3805

**Pull request closed without merging**